### ISSUE TITLE
Calling getProperty with explicit Map type returns org.grails.config.NavigableMap

### ISSUE THREAD

**Issue by puneetbehl** [grails/grails-core#12415]
### Expected Behavior

When we call `grailsApplication.config.getProperty("foo", Map.class)`, then it should not return a `org.grails.config.NavigableMap` where _application.yml_ contains following:

```
foo:
  username: test
  password: secret
```

### Actual Behaviour

It appears that the call to`grailsApplication.config.getProperty("foo", Map.class)` return a NavigableMap and also display deprecated usage warnings which is wrong.

### Steps To Reproduce

1. Create a new Grails application
2. Add following to the _grails-app/init/myapp/BootStrap.groovy_
```
    GrailsApplication grailsApplication

    def init = { servletContext ->
        Map appInfo = grailsApplication.config.getProperty("info.app", Map)
        !(appInfo instanceof NavigableMap)
    }
```
3. Execute application using `./gradlew bootRun`

### Environment Information

- Operating System: macOS
- JDK Version: 1.8

### Version

5.1.2

**Comment by mhuebner** [comment]
Any chance to get this into a hotfix release?

**Comment by puneetbehl** [comment]
This should be resolved with 5.0.3
